components/layout/GenericPostLayout.tsx

```diff
@@ -0,0 +1,60 @@
+import ErrorPage from 'next/error';
+import { useRouter } from 'next/router';
+
+import type { IPosts } from '@/types/post';
+
+import GenericPostContext from '../../context/GenericPostContext';
+import AnnouncementHero from '../campaigns/AnnoucementHero';
+import Head from '../Head';
+import Container from './Container';
+
+interface IGenericPostLayoutProps {
+  // Commnon to all posts
+  post: IPosts['blog'][number];
+  children: React.ReactNode;
+}
+
+/**
+ * @param props.post - The blog post to render in the layout
+ * @param props.children - The content of the blog post
+ * @description The blog layout with the post and its content
+ */
+export default function GenericPostLayout({ post, children } : IGenericPostLayoutProps) {
+  const router = useRouter();
+
+  if (!post) return <ErrorPage statusCode={404} />;
+  if (post.title === undefined && post.slug !== '/about') throw new Error('Post title is required');
+
+  if (!router.isFallback && !post?.slug) {
+    return <ErrorPage statusCode={404} />;
+  }
+
+  return (
+    <GenericPostContext.Provider value={{ post }}>
+      <AnnouncementHero className='m-4 text-center' small={true} />
```

Comment:
You can comment out this line and better exclude the campaigns components from this PR.